default_dialer.go

```diff
@@ -0,0 +1,49 @@
+package enigma
+
+import (
+	"context"
+	"github.com/gorilla/websocket"
+	"github.com/pkg/errors"
+	"net"
+	"net/http"
+)
+
+func setupDefaultDialer(dialer *Dialer) {
```

Comment:
I am wondering if it could be better to have a default CreateSocket function plainly declared in the global scope. Is there any specific rationale as to why it is an anonymous function? We could have something like:
```go
func defaultCreateSocket(ctx context.Context, url string, httpHeader http.Header) (Socket, error) {...}
```
So we can write:
```go
if dialer.CreateSocket == nil {
    dialer.CreateSocket = defaultCreateSocker
}
```
or something like that?